api/logs/src/main/java/io/opentelemetry/api/events/EventLogger.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.events;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.logs.LogRecordBuilder;
+import io.opentelemetry.api.logs.Logger;
+import io.opentelemetry.api.logs.Severity;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link EventLogger} provides convenience methods for emitting Events. An Event is a {@code
+ * LogRecord} that conform to the <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/semantic_conventions/events.md">Event
+ * Semantic Conventions</a>.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * class MyClass {
+ *   private final LoggerProvider loggerProvider = GlobalLoggerProvider.get();
+ *   // Create EventLogger by wrapping Logger and setting event domain to "acme-observability"
+ *   private final EventLogger eventLogger =
+ *     EventLogger.create(
+ *       loggerProvider.get("instrumentation-library-name"), "acme.observability");
+ *
+ *   void doWork() {
+ *     eventLogger
+ *       .eventBuilder("my-event")
+ *       .setAllAttributes(Attributes.builder()
+ *          .put("key1", "value1")
+ *          .put("key2", "value2").build())
+ *       .emit();
+ *   }
+ * }
+ * }</pre>
+ */
+public final class EventLogger {
```

Comment:
I think it would be very useful to additionally have a convenience method to emit directly with an event name and attributes, rather than forcing to go through a builder. In fact, I might start with _just_ that, and only add the builder if we have an ask for it. Thoughts?

Reply:
I think we'll likely want to tweak the Event API surface area to cater to Event use cases. Making that change now is simultaneously conservative and speculative. Conservative because the surface area is smaller than with a builder. Speculative because as it stands now in the spec, Events should be able to set all the fields of `LogRecord`. Though, there's [reason](https://github.com/open-telemetry/opentelemetry-specification/pull/2941#discussion_r1020527045) to believe this could change in favor of the direction you're suggesting.

Some additional context: this [PR](https://github.com/open-telemetry/opentelemetry-specification/pull/2926) suggests encapsulating event specific attributes in an `event.data` attribute (which would have to be implemented as a map). [This conversation](https://github.com/open-telemetry/opentelemetry-specification/pull/2926#discussion_r1016950931) suggests instead to put the event data into the `body` field. If event data is encapsulated in some field, there's also been discussions of some sort of description of some indication of "content type". The purpose being that most of the time event data is key value pairs that can be represented as attributes, but maybe there are use cases for it to be binary data instead. 🤷 Sharing these conversations to get folks thinking about how the API surface area might accommodate these changes, if they do in fact come to fruition. 

Overall, I'm in favor limiting the API surface area for now, and waiting for feedback from users to expand to additional fields. So we'd have:
```
public class EventLogger {
  ...
  void emitEvent(String eventName, Attributes attributes) { ... }
  
  // Can add a builder or emitEvent overloads later if needed based on demand and patterns that emerge from the specification
  // EventBuilder eventBuilder(String eventName)
}
```